beets/dbcore/db.py

```diff
@@ -594,20 +604,31 @@ def __iter__(self):
             # Objects are pre-sorted (i.e., by the database).
             return self._get_objects()
 
-    def _make_model(self, row):
-        # Get the flexible attributes for the object.
+    def _get_flex_attrs(self, ids):
+        # Get the flexible attributes for all ids.
         with self.db.transaction() as tx:
+            id_list = ', '.join(str(id) for id in ids)
             flex_rows = tx.query(
-                'SELECT * FROM {0} WHERE entity_id=?'.format(
-                    self.model_class._flex_table
-                ),
-                (row['id'],)
+                'SELECT * FROM {0} WHERE entity_id IN ({1})'.format(
+                    self.model_class._flex_table,
+                    id_list
+                )
             )
 
+        # Index flexible attributes by the entity id they belong to
+        flex_values = dict()
+        for row in flex_rows:
+            if row['entity_id'] not in flex_values:
+                flex_values[row['entity_id']] = dict()
+
+            flex_values[row['entity_id']][row['key']] = row['value']
+
+        return flex_values
+
+    def _make_model(self, row, flex_values={}):
```

Comment:
As above, a docstring would be nice here (even though we didn't have one before). It's also worth noting that the contract for this method is now different than it was: you need to have already fetched the flexattrs; previously the method did it for you.

Reply:
I considered `_make_model` a private method, so breaking the contract didn't seem like that big a deal. I can figure out a way to keep the current API in tact, but not sure it's worth it.

Reply:
Yeah, absolutely—I agree it's no big deal to change the interface here. But just like the new method above, it would be nice to have a docstring about what it's supposed to do now so we can be aware of the difference in the future.